### PR TITLE
New Module: win_feature_info

### LINKED ISSUE #208: Windows feature info module
##### SUMMARY
Create a new module win_feature_info

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
```yaml
- name: get windows feature info
  community.windows.win_feature_info:
    name: RDS-Licensing
  register: feature_info

- name: get windows feature info for all features
  community.windows.win_feature_info:
  register: feature_info
```
##### ADDITIONAL INFORMATION
To provide information on the current state of a windows feature.
This is needed for queries about system state as well for making decisions about items like mounting the sxs folder so unavailable features can be added.

```yaml
- name: get windows feature info
  community.windows.win_feature_info:
    name: RDS-Licensing
  register: feature_info

- name: Install License Role
  ansible.windows.win_feature:
    name: RDS-Licensing
    state: present
  register: licensing_install
  when: feature_info.features[0].install_state == "Available"

- name: Install License Role
  ansible.windows.win_feature:
    name: RDS-Licensing
    state: present
    source: C:\temp\sxs\
  register: licensing_install
  when: feature_info.features[0].install_state == "Removed"
```

### PR DESCRIPTION
**Pull request by gamethis**
##### SUMMARY
New module win_feature_info

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
win_feature_info

##### ADDITIONAL INFORMATION
closes #208

### CODE REVIEW

File: plugins/modules/win_feature_info.ps1 [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+$name = $module.Params.name
+
+$module.Result.exists = $false
+
+$features = Get-WindowsFeature -Name $name
+
+$module.Result.features = @(foreach ($feature in ($features)) {
+        # These should closely reflect the options for win_feature
+        [Ordered]@{
```
Comment (jborean93) [inline plugins/modules/win_feature_info.ps1:26]: Minor nit, there's not much point in using `[Ordered]` here. The keys are sorted alphabetically on the Ansible side and there is never a guarantee that the order will stay the same from PowerShell's perspective.

File: plugins/modules/win_feature_info.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_feature_info
+short_description: Gather information about Windows features
```
Comment (jborean93) [inline plugins/modules/win_feature_info.py:10]: You need to add `version_added: '1.4.0'` (1.4.0 is the next release) to the documentation to denote what version this was added in.

File: plugins/modules/win_feature_info.ps1 [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#!powershell
+
+# Copyright: (c) 2020, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#AnsibleRequires -CSharpUtil ansible_collections.ansible.windows.plugins.module_utils.SCManager
```
Comment (jborean93) [inline plugins/modules/win_feature_info.ps1:7]: You don't need this one here.

**Review by jborean93 — CHANGES_REQUESTED** [review]
Sorry for the delay in reviewing this, the additions look great. The only thing we need to see is some tests. This can just be a simple test for checking exists is set to True and False for the respective scenarios.

File: plugins/modules/win_feature_info.ps1 (hunk 1 above)
Comment (gamethis) [reply plugins/modules/win_feature_info.ps1]: completed

**Comment by gamethis** [comment]
I made the changes you suggested.  I got some tests in... not sure why it is failing.  could use some help on this @jborean93

**Comment by gamethis** [comment]
> I made the changes you suggested. I got some tests in... not sure why it is failing. could use some help on this @jborean93

Nevermind... I figured out what was wrong... forgot to update a file to match my local test copy.

**Comment by gamethis** [comment]
> I made the changes you suggested. I got some tests in... not sure why it is failing. could use some help on this @jborean93

Nevermind... I figured out what was wrong... forgot to update a file to match my local test copy.

**Review by jborean93 — APPROVED** [review]
Looks great, thanks for contributing this.

**Pull request merged**